share/spack/setup-env.sh

```diff
@@ -236,7 +236,18 @@ export SPACK_ROOT=${_sp_prefix}
 # Determine which shell is being used
 #
 function _spack_determine_shell() {
-	PS_FORMAT= ps -p $$ | tail -n 1 | awk '{print $4}' | sed 's/^-//' | xargs basename
+	# This logic is derived from the cea-hpc/modules profile.sh example.
+	# The objective is to correctly detect shell, even when setup-env is 
+	# sourced within a script itself, rather than a login terminal. 
+	if [ -n "${BASH:-}" ]; then
+		echo ${BASH##*/}
+	elif [ -n "${ZSH_NAME:-}" ]; then
+		echo $ZSH_NAME
+	else
+		PS_FORMAT= ps -p $$ | tail -n 1 | awk '{print $4}' | sed 's/^-//' | xargs basename
+		# alternative to above:
+		# basename $(ps -p $$ -ocomm=)
```

Comment:
@carns I'm not sure what the benefit of listing an alternative is as long as this is working (which it normally has, minus the case of executing inside of a script)

Also, I think it may be worthwhile to include a link to `https://github.com/cea-hpc/modules/blob/master/init/profile.sh.in` directly.

Reply:
Agreed, thanks @scheibelp !